### PR TITLE
refactor: Signout modal

### PR DESCRIPTION
**Pull request by gee05053**
This PR resolves https://github.com/lablup/backend.ai-webui/issues/1965

Before
스크린샷 2023-12-10 오전 3 19 03

After
스크린샷 2023-12-12 오전 9 54 50

**Checklist:** (if applicable)

- [x] Mention to the original issue
- [ ] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

**Review by agatha197 — CHANGES_REQUESTED** [review]
I think you'd better to use `webui.menu.<something>` to translate. Or remove the original signout-related translated words from the i18n json files.

File: react/src/components/SignoutModal.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        width={450}
+        open={open}
+        onOk={handleOk}
+        okText={t('webui.menu.LeaveService')}
+        onCancel={() => {
+          onRequestClose();
+        }}
+        {...modalProps}
+      >
+        <Form form={form} labelCol={{ span: 6 }}>
```
Comment (yomybaby) [inline react/src/components/SignoutModal.tsx]: If there is no specific reason, please use the vertical layout as default.

**Review by yomybaby — CHANGES_REQUESTED** [review]

File: react/src/components/SignoutModal.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  };
+  return (
+    <>
+      <BAIModal
+        title={t('webui.menu.LeaveService')}
+        centered
+        width={450}
+        open={open}
+        onOk={handleOk}
+        okText={t('webui.menu.LeaveService')}
```
Comment (yomybaby) [inline react/src/components/SignoutModal.tsx]: Please set the danger style to the `LeaveService` button because it is a destructive action.

**Review by yomybaby — CHANGES_REQUESTED** [review]

**Review by lizable — APPROVED** [review]
LGTM.

File: react/src/components/SignoutModal.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        <Form form={form} layout="vertical" labelCol={{ span: 6 }}>
+          <Form.Item name="alert">
+            <Alert message={t('login.DescConfirmLeave')} type="warning" />
+          </Form.Item>
+          <Form.Item
+            name="email"
+            label={t('general.E-Mail')}
+            required
+            rules={[
+              () => ({
```
Comment (agatha197) [inline react/src/components/SignoutModal.tsx:82]:
```suggestion
              { type: 'email' }
              () => ({
```

**Review by agatha197 — CHANGES_REQUESTED** [review]
Let's add email format validation or you can check if the email value matches with `baiClient.email`.

File: react/src/components/SignoutModal.tsx (hunk 3 above)
Comment (yomybaby) [reply react/src/components/SignoutModal.tsx:82]: @gee05053 please check this review and apply.

Comment (gee05053) [reply react/src/components/SignoutModal.tsx:82]: @yomybaby I resolved this review now!

**Review by yomybaby — CHANGES_REQUESTED** [review]
Please remove deprecated previous signout-panel and related code such as [this code](https://github.com/lablup/backend.ai-webui/blob/832ef4d0933abd294ec642eddca3063d56cf2c44/src/components/backend-ai-login.ts#L2318-L2360)

**Comment by yomybaby** [comment]
Is there any reason for changing from '서비스 탈퇴' to '탈퇴'? @gee05053

**Pull request merged**